system/Security/Security.php

```diff
@@ -318,24 +318,30 @@ private function removeTokenInRequest(RequestInterface $request): void
     {
         assert($request instanceof Request);
 
-        $json = json_decode($request->getBody() ?? '');
-
         if (isset($_POST[$this->config->tokenName])) {
             // We kill this since we're done and we don't want to pollute the POST array.
             unset($_POST[$this->config->tokenName]);
             $request->setGlobal('post', $_POST);
-        } elseif (isset($json->{$this->config->tokenName})) {
-            // We kill this since we're done and we don't want to pollute the JSON data.
-            unset($json->{$this->config->tokenName});
-            $request->setBody(json_encode($json));
+        } else {
+            $body = $request->getBody() ?? '';
+            if (! empty($json = json_decode($body)) && json_last_error() === JSON_ERROR_NONE) {
```

Comment:
can `$json = json_decode($body)` moved before if condition like original code?

Reply:
I see no point. If the CSRF token is placed in the POST array, then we're just wasting the resources. Is there any particular reason for what you're proposing? Because that could change my view.

Reply:
nothing, just preference. placing before if doesn't wast resource, as the condition is executed in before `&&` so always executed, also `json_decode()` doesn't use `THROW_ON_ERROR` so it doesnt' cause side effect with then rely on `json_last_error()` result.

Reply:
Oh... sorry, I think I get what you mean. The "like original code" part misled me. Something like this below?
```suggestion
            $json = json_decode($body);
            if (! empty($json) && json_last_error() === JSON_ERROR_NONE) {
```

Reply:
Yes 👍

Reply:
Great - done! Thanks!